src/main/scala/com.snowplowanalytics.sauna/observers/AmazonKinesisObserver.scala

```diff
@@ -0,0 +1,121 @@
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.sauna
+package observers
+
+// java
+import java.util.Date
+
+// scala
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+// akka
+import akka.actor._
+
+// amazonaws
+import com.amazonaws.auth.{AWSCredentialsProvider, BasicAWSCredentials}
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.{InitialPositionInStream, KinesisClientLibConfiguration}
+import com.amazonaws.services.kinesis.model.Record
+
+// kinesis
+import com.gilt.gfc.aws.kinesis.client.{KCLConfiguration, KCLWorkerRunner, KinesisRecordReader}
+
+// sauna
+import Observer.KinesisRecordReceived
+
+class AmazonKinesisObserver(streamName: String, kclConfig: KinesisClientLibConfiguration) extends Actor with Observer {
+  override def preStart: Unit = {
+    super.preStart()
+    notifyLogger("Started Kinesis Observer")
+
+    /**
+     * [[KinesisRecordReader]] instance that will return processed records
+     * without any transformation.
+     */
+    implicit object ARecordReader extends KinesisRecordReader[Record] {
+      override def apply(r: Record): Record = {
+        r
+      }
+    }
+
+    /**
+     * Start a worker thread that will notify the mediator about processed
+     * Kinesis records.
+     */
+    KCLWorkerRunner(kclConfig).runAsyncSingleRecordProcessor[Record](1 minute) { record: Record =>
+      Future {
+        notifyLogger(s"Received Kinesis Record from $streamName")
+        context.parent ! KinesisRecordReceived(streamName, record.getSequenceNumber, record.getData, self)
+      }
+    }
+  }
+
+  def receive: Receive = {
+    case _ =>
+  }
+}
+
+object AmazonKinesisObserver {
+  def props(streamName: String, kclConfig: KinesisClientLibConfiguration): Props =
+    Props(new AmazonKinesisObserver(streamName, kclConfig))
+
+  def props(config: AmazonKinesisConfig_1_0_0): Props = {
+
+    // AWS configuration. Safe to throw exception on initialization
+    val credentials = new BasicAWSCredentials(
+      config.parameters.aws.accessKeyId,
+      config.parameters.aws.secretAccessKey)
+
+    class KinesisCredentialsProvider(credentials: BasicAWSCredentials) extends AWSCredentialsProvider {
+      def refresh(): Unit = {}
+
+      def getCredentials: BasicAWSCredentials = credentials
+    }
+
+    val credentialsProvider = new KinesisCredentialsProvider(credentials)
```

Comment:
The SDK resolves as aws-java-sdk-core:1.1.18 (from https://github.com/gilt/gfc-aws-kinesis), but that class was added in 1.1.30.